### PR TITLE
[e2e] vtctld init tablet and some output-based commands

### PR DESCRIPTION
**Pull request by ajm188**
## Description

This PR began as an attempt to remove the deprecated `InitTablet` with the underlying topo call, and then evolved to also migrate some of the output-based commands from `vtctlclient` to `vtctldclient`

## Related Issue(s)

https://github.com/vitessio/vitess/issues/15273

## Checklist

-   [x] "Backport to:" labels have been added if this change should be back-ported to release branches
-   [x] If this change is to be back-ported to previous releases, a justification is included in the PR description
-   [x] Tests were added or are not required
-   [x] Did the new or modified tests pass consistently locally and on CI?
-   [x] Documentation was added or is not required

### CODE REVIEW

File: go/test/endtoend/cluster/cluster_process.go
Comment (deepthi) [inline go/test/endtoend/cluster/cluster_process.go:248]: How was this working earlier? If we weren't starting the topo at all..

Comment (deepthi) [inline go/test/endtoend/cluster/cluster_process.go:248]:
```suggestion
	if err = cluster.TopoProcess.OpenServer(*topoFlavor, cluster.VtctldClientProcess.TopoGlobalRoot); err != nil {
```

File: go/test/endtoend/cluster/topo_process.go
Comment (deepthi) [inline go/test/endtoend/cluster/topo_process.go:82]: why is this block needed?

**Review by deepthi — APPROVED** [review]
Nice! A couple of little things need fixing, rest LGTM.

File: go/test/endtoend/cluster/cluster_process.go
Comment (ajm188) [reply go/test/endtoend/cluster/cluster_process.go:248]: never needed to directly, because everything would happen through the `VtctlProcess`, which makes its own call to `topo.Open`, or one of the clients.

Comment (ajm188) [reply go/test/endtoend/cluster/cluster_process.go:248]: Does not exist (note the original was `VtctlProcess` _not_ `VtctlclientProcess`)

File: go/test/endtoend/cluster/topo_process.go
Comment (ajm188) [reply go/test/endtoend/cluster/topo_process.go:82]: relic from when I was calling `topo.Open` in this function, i'll undo this bit

File: go/test/endtoend/keyspace/keyspace_test.go
Comment (mattlord) [inline go/test/endtoend/keyspace/keyspace_test.go:265]: Why not use the `ShardReplicationPositions` vtctldclient command here and below? Is it not equivalent?

File: go/test/endtoend/tabletmanager/commands_test.go
Comment (mattlord) [inline go/test/endtoend/tabletmanager/commands_test.go:26]: Nit, but we could clean up the imports here.

**Review by mattlord — COMMENTED** [review]
LGTM! I only had some minor comments. Let me know what you think and I'll come back to this quickly.

File: go/test/endtoend/keyspace/keyspace_test.go
Comment (ajm188) [reply go/test/endtoend/keyspace/keyspace_test.go:265]: 🤦 simply me grepping too exactly (started with "Get")

**Comment by ajm188** [comment]
idk why it's not letting me reply inline to https://github.com/vitessio/vitess/pull/15297#discussion_r1506367328 but 

> Without also removing the vtctl binary as well we can't get rid of any of the vtctlclient related code so I think it's important. Unless you think I'm missing something?

That's absolutely correct. Let me rework the GlobalRoot to set it somewhere in addition to `VtctlProcess`

File: go/test/endtoend/keyspace/keyspace_test.go
Comment (ajm188) [reply go/test/endtoend/keyspace/keyspace_test.go:265]: added in 2014 as part of ["more test cleanup"](https://github.com/vitessio/vitess/commit/8f608994821c6eb7ade6fa79574d96ab65546cfb) with no further details (committed directly to main, no PR).

Looking at the [message definition](https://github.com/vitessio/vitess/blob/main/proto/topodata.proto#L305-L315), it's not ... super instructive? From the [usage of the topo method](https://github.com/search?q=repo%3Avitessio%2Fvitess+%22.GetShardReplication%22+NOT+language%3A%22Protocol+Buffer%22+NOT+path%3A%2F%5Ego%5C%2Fvt%5C%2Fvtadmin%5C%2F%2F+NOT+path%3A%2F%5Eweb%5C%2F%2F&type=code), it's just used to get a list of tablets in a shard (without needing to start at the shard primary, and then walk the replication graph manually).

This is updated during tablet init via [`UpdateTabletReplicationData`](https://github.com/vitessio/vitess/blob/15f58869c45e47875f26e95291bcc656a686a269/go/vt/topo/tablet.go#L506) and conversely with [`DeleteTabletReplicationData`](https://github.com/vitessio/vitess/blob/15f58869c45e47875f26e95291bcc656a686a269/go/vt/topo/tablet.go#L511) to remove a node.

Given that we include `ShardReplication{Add,Fix,Remove}` to manually edit the topo data, it would be useful to "get" that data to inspect it before editing, so I will add the command to the new client.

Comment (ajm188) [reply go/test/endtoend/keyspace/keyspace_test.go:265]: See https://github.com/vitessio/vitess/pull/15389, which I will rebase this on after merging.

**Comment by ajm188** [comment]
@mattlord no rush at all, but this is ready for another review whenever you get some time :)

File: go/test/endtoend/recovery/pitr/shardedpitr_test.go
Comment (mattlord) [inline go/test/endtoend/recovery/pitr/shardedpitr_test.go:590]: Not sure it's even worth a CI run, but response is misspelled.

File: go/test/endtoend/reparent/utils/utils.go
Comment (mattlord) [inline go/test/endtoend/reparent/utils/utils.go:621]: We can't use `vtctldclient ShardReplicationPositions` instead?

File: go/test/endtoend/tabletgateway/buffer/reshard/sharded_buffer_test.go
Comment (mattlord) [inline go/test/endtoend/tabletgateway/buffer/reshard/sharded_buffer_test.go:51]: responce creeping around again :-)

File: go/test/endtoend/tabletgateway/vtgate_test.go
Comment (mattlord) [inline go/test/endtoend/tabletgateway/vtgate_test.go:127]: Curious why we're using `.ChangeTabletType()` in one place and `.ExecuteCommand()` in the other.

**Review by mattlord — APPROVED** [review]
LGTM. Thank you for working on this, @ajm188 !

**Pull request merged**
